Patch to make it compatible with newest NameLayer

suirad opened the pull request:
I had to make these changes to make it compatible with the latest NameLayer. Were there changes to it? Looks like the command stuff was moved from namelayer to civmodcore; at least from a civchat perspective. @rourke750 
This is the error i was getting:
![Error img](http://gyazo.com/ec9d7edb27ea55a145dc8ff957805a79.png)

suirad committed:
Patch to make it compatible with newest NameLayer

suirad commented:
Looks like Civchat needs a CLA

ttk2 commented:
Civchat didn't get its CLA yet? oh well we can get that up soon, you agree to MIT? 

Anyways Namelayer has been changed to move to Civmodcore on the backend its not really worked out at all yet, so we need to sit down and move all of this online which will be a near term priority.

suirad commented:
Yup I agree.
Yea thats fine. Just means mods that rely on some parts of namelayer need to be modified. This patch was pretty simple, just needed to be done before i try to add mercury.

suirad committed:
Another push. I am actually having an issue here. see comment in the code.

ttk2 commented:
make a post to the dev group, maybe someone there can help, roruke will probably be by soon enough.

suirad commented:
I added the method here:
4223e76

rourke750 commented on src/vg/civcraft/mc/civchat2/listeners/MercuryMessageListener.java:
```diff
@@ -0,0 +1,48 @@
+import vg.civcraft.mc.namelayer.NameAPI;
+import vg.civcraft.mc.namelayer.NameLayerPlugin;
+
+public class MercuryMessageListener extends BukkitRunnable implements Listener {
```
This class does not need to be a runnable.  Just make it implement Listener and register it in the main class.  The issue in namelayer was that namelayer is programmed to try and be the first program running always.  Civchat should handle its dependencies fine since Mercury for sure will load before it.

suirad replied:
Alright cool il test it out

rourke750 replied:
Also on a side note.  BukkitRunnable is depricated and they suggest just using a regular Runnable

ttk2 commented:
is this good otherwise?

rourke750 commented:
@ttk2 not yet

rourke750 commented:
@ttk2 Oh besides those things yea this is fine.

ttk2 commented:
ok then will merge once those are fixed.

suirad commented:
I figured out what my issue was, i forgot to remove namelayer from the pom
of civchat before i got started, so when i ran maven it pulled it as the
namelayer dependancy and Thats why it wasnt showing any changes of methods
i had added or removed. Other than that i didn't have a lot of time to work
on it, but i should have this finished soon. Il
On Jun 6, 2015 10:53 PM, "ttk2" notifications@github.com wrote:

> ok then will merge once those are fixed.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/Civcraft/CivChat2/pull/12#issuecomment-109643672.

ttk2 commented:
Cool. Just tell me when its ready. Thanks!

On Sat, Jun 6, 2015, 3:31 PM Darius Scott notifications@github.com wrote:

> I figured out what my issue was, i forgot to remove namelayer from the pom
> of civchat before i got started, so when i ran maven it pulled it as the
> namelayer dependancy and Thats why it wasnt showing any changes of methods
> i had added or removed. Other than that i didn't have a lot of time to work
> on it, but i should have this finished soon. Il
> On Jun 6, 2015 10:53 PM, "ttk2" notifications@github.com wrote:
> 
> > ok then will merge once those are fixed.
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > https://github.com/Civcraft/CivChat2/pull/12#issuecomment-109643672.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/Civcraft/CivChat2/pull/12#issuecomment-109646330.

rourke750 commented:
@suirad Still working on this?

suirad committed:
Confirmed working cross server pms

suirad commented:
Yup just put out another pull. This one is currently working pms. Of course it needs to be cleaned up some and improved a bit for other civchat features, but testable pms nonetheless.

suirad commented:
Im done for tonight. Tomorrow i plan on doing group chat. Just to be clear, group messaging, pms, and what else needs to be done for civchat to be shard ready? 

Also right now a lot of the messages through civchat are broadcast to all servers. This is based off of namelayer only keeping track of just a list of players across all other servers. I think i may change it to also keep track of what named server each player is on, so that civchat can request that and use directed messages for pms and group chat. 

@ttk2  as of right now, i would like to ask about any concerns you have for current and future devs working with mercury and venus. As of right now, most code i have committed for both namelayer and civchat regarding mercury use is mostly contained to the Classes "MercuryMessageListener(receives messages) and the few sprinkled parts around the code that fires off messages. Also in both plugins, mercury parts are considered optional. 

My question is would you agree this is ideal for continued use of Mercury & Venus.

rourke750 commented:
@suirad If you could make global chat configurable as its not something we would really want, optionally maybe send the cords of where the broadcast came from so depending on where they are they can still hear the message if they are within the chat range but on a different shard.

Also as long as all calls are solely in Mercury/ NameLayer and no direct calls to Venus everything should be fine.

rourke750 commented:
ok to test

suirad commented:
@rourke750 Does civchat already support configurable global chat? If so, i could just start there and inject a broadcast message after it is sent to everyone on the initial server.

rourke750 commented:
@suirad It does look here: https://github.com/Civcraft/CivChat2/blob/master/src/vg/civcraft/mc/civchat2/CivChat2Manager.java#L213

suirad commented:
@rourke750 Okay il start working there when i do it. I can make global chat working, easily. Im not exactly sure how you will want coord chat to work xserver though. Unless you just want it to just do literal coords,  aka same coords on other servers.

rourke750 commented:
@suirad Oh that's a good point.  Add a config option so we can give each chat world a name and have similar worlds have same chat names.  Then pass the chat name of the world and coords and each server will handle it on its own.

I don't want to do it based on the world name as it will mess other sharding stuff up as each world file name will need to be unique.

jjj5311 commented:
@suirad  Dont worry about cross server chat @ttk2 said he wasn't worried about shard-border chat from here https://github.com/Civcraft/CivChat/issues/9

jjj5311 commented:
Also if you want some help with this just let me know what you want done

ttk2 commented:
yes, ignore cross border chat, concern yourself with PM's only. 

As for how Mercury/Venus will be used I want to get another dev to try and do a Mercury conversion for a plugin before I make comments, I would like their feedback on whats good and whats bad about the way its setup now.

suirad commented:
@ttk2 how about configurable xserver global chat? 

Alright cool, let me know whom you have to do it, so i can look for any questions they have.

ProgrammerDan commented:
@suirad IMHO cross shard chat, preserving ranging (e.g. configurable xserver global chat), is critical _long term_, although it is definitely not important for first deliverable.

suirad commented:
Doing it is fine, the question is how to implement.

ProgrammerDan commented:
Definitely. I'd recommend just making it a top-level issue and coming back to it later, assuming the infrastructural support is largely there (an approximate path-to-solution is evident).

ttk2 commented:
Dan. I explicitly don't want cross shard chat for various reasons that we
talked about in the admin group way back. Probably need to replicate that
discussion in public.

Cross server global is fine if you feel like making it.

On Tue, Jun 9, 2015, 10:25 AM Daniel Boston notifications@github.com
wrote:

> Definitely. I'd recommend just making it a top-level issue and coming back
> to it later, assuming the infrastructural support is largely there (an
> approximate path-to-solution is evident).
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/Civcraft/CivChat2/pull/12#issuecomment-110401963.

ttk2 commented:
What else needs doing here suirdax do you need anyone to help?

On Tue, Jun 9, 2015, 5:50 PM justin kilpatrick kilpatrickjustin@gmail.com
wrote:

> Dan. I explicitly don't want cross shard chat for various reasons that we
> talked about in the admin group way back. Probably need to replicate that
> discussion in public.
> 
> Cross server global is fine if you feel like making it.
> 
> On Tue, Jun 9, 2015, 10:25 AM Daniel Boston notifications@github.com
> wrote:
> 
> > Definitely. I'd recommend just making it a top-level issue and coming
> > back to it later, assuming the infrastructural support is largely there (an
> > approximate path-to-solution is evident).
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > https://github.com/Civcraft/CivChat2/pull/12#issuecomment-110401963.

jjj5311 commented:
I think we were referring to cross shard global chat, are you saying you
don't want cross shard chat for group messaging/private messaging?

On Tuesday, June 9, 2015, ttk2 notifications@github.com wrote:

> Dan. I explicitly don't want cross shard chat for various reasons that we
> talked about in the admin group way back. Probably need to replicate that
> discussion in public.
> 
> Cross server global is fine if you feel like making it.
> 
> On Tue, Jun 9, 2015, 10:25 AM Daniel Boston <notifications@github.com
> <javascript:_e(%7B%7D,'cvml','notifications@github.com');>>
> wrote:
> 
> > Definitely. I'd recommend just making it a top-level issue and coming
> > back
> > to it later, assuming the infrastructural support is largely there (an
> > approximate path-to-solution is evident).
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > https://github.com/Civcraft/CivChat2/pull/12#issuecomment-110401963.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/Civcraft/CivChat2/pull/12#issuecomment-110526389.

ttk2 commented:
We want it for groups and pm's but not like area chat in adjacent shards.

On Tue, Jun 9, 2015, 5:52 PM jjj5311 notifications@github.com wrote:

> I think we were referring to cross shard global chat, are you saying you
> don't want cross shard chat for group messaging/private messaging?
> 
> On Tuesday, June 9, 2015, ttk2 notifications@github.com wrote:
> 
> > Dan. I explicitly don't want cross shard chat for various reasons that we
> > talked about in the admin group way back. Probably need to replicate that
> > discussion in public.
> > 
> > Cross server global is fine if you feel like making it.
> > 
> > On Tue, Jun 9, 2015, 10:25 AM Daniel Boston <notifications@github.com
> > <javascript:_e(%7B%7D,'cvml','notifications@github.com');>>
> > wrote:
> > 
> > > Definitely. I'd recommend just making it a top-level issue and coming
> > > back
> > > to it later, assuming the infrastructural support is largely there (an
> > > approximate path-to-solution is evident).
> > > 
> > > —
> > > Reply to this email directly or view it on GitHub
> > > https://github.com/Civcraft/CivChat2/pull/12#issuecomment-110401963.
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > https://github.com/Civcraft/CivChat2/pull/12#issuecomment-110526389.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/Civcraft/CivChat2/pull/12#issuecomment-110526614.

ProgrammerDan commented:
@ttk2 I'd very strongly suggest making that discussion public, because it seems like a major feature regression. Part of the "adventure" of Civcraft is finding you are near people. Given the focus on portal based sharding, I can guess your reasons; I find _that_ to be an incredible regression for reasons I've begun elucidating in the discourse. So ... If portal based sharding, I've no objection to not sending general chat across shards. If "smooth" sharding, it'd be ridiculous not to have it.

ttk2 commented:
I think that trying to smooth sharding too much will make things harder.

Cross server pm's and group messages are a must. But area chat leaking over
is not essential ihmo. Too complex to manage all the time.

On Tue, Jun 9, 2015, 6:00 PM Daniel Boston notifications@github.com wrote:

> @ttk2 https://github.com/ttk2 I'd very strongly suggest making that
> discussion public, because it seems like a major feature regression. Part
> of the "adventure" of Civcraft is finding you are near people. Given the
> focus on portal based sharding, I can guess your reasons; I find _that_
> to be an incredible regression for reasons I've begun elucidating in the
> discourse. So ... If portal based sharding, I've no objection to not
> sending general chat across shards. If "smooth" sharding, it'd be
> ridiculous not to have it.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/Civcraft/CivChat2/pull/12#issuecomment-110527721.

jjj5311 commented:
totally off topic can someone sponsor me on discourse    name=jjj5311

ghost commented:
You've contributed code so not necessary. What is your discourse name?
On Jun 9, 2015 8:57 PM, "jjj5311" notifications@github.com wrote:

> totally off topic can someone sponsor me on discourse name=jjj5311
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/Civcraft/CivChat2/pull/12#issuecomment-110549034.

jjj5311 commented:
same as on here jjj5311

On Tue, Jun 9, 2015 at 7:59 PM, Dr. Andrew Jawa notifications@github.com
wrote:

> You've contributed code so not necessary. What is your discourse name?
> On Jun 9, 2015 8:57 PM, "jjj5311" notifications@github.com wrote:
> 
> > totally off topic can someone sponsor me on discourse name=jjj5311
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > https://github.com/Civcraft/CivChat2/pull/12#issuecomment-110549034.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/Civcraft/CivChat2/pull/12#issuecomment-110549256.

ghost commented:
It is done
On Jun 9, 2015 9:00 PM, "jjj5311" notifications@github.com wrote:

> same as on here jjj5311
> 
> On Tue, Jun 9, 2015 at 7:59 PM, Dr. Andrew Jawa notifications@github.com
> wrote:
> 
> > You've contributed code so not necessary. What is your discourse name?
> > On Jun 9, 2015 8:57 PM, "jjj5311" notifications@github.com wrote:
> > 
> > > totally off topic can someone sponsor me on discourse name=jjj5311
> > > 
> > > —
> > > Reply to this email directly or view it on GitHub
> > > https://github.com/Civcraft/CivChat2/pull/12#issuecomment-110549034.
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > https://github.com/Civcraft/CivChat2/pull/12#issuecomment-110549256.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/Civcraft/CivChat2/pull/12#issuecomment-110549371.

suirad commented:
@ttk2 all thats left is to add group chat. For pms i need to make /r or the
reply class work fine sharded. I also need to go to pms themselves and make
sure it checks for mercury enabled before trying to do shard stuff.

Question about the Civcraft repos. Would it be fine to make issues on there
as notes for things i need to add/fix so i don't forget. I went to put an
issue on my own namelayer repo but there is no issues section for mine.

ghost commented:
You have to enable issues in the repo setting.
On Jun 9, 2015 11:40 PM, "Darius Scott" notifications@github.com wrote:

> @ttk2 all thats left is to add group chat. For pms i need to make /r or the
> reply class work fine sharded. I also need to go to pms themselves and make
> sure it checks for mercury enabled before trying to do shard stuff.
> 
> Question about the Civcraft repos. Would it be fine to make issues on there
> as notes for things i need to add/fix so i don't forget. I went to put an
> issue on my own namelayer repo but there is no issues section for mine.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/Civcraft/CivChat2/pull/12#issuecomment-110575937.

suirad commented:
Figured it was something simple.

ttk2 commented:
Yes issues for reminders are fine. Just tag with the lower priority tags.

Good to hear it's close. Let me know if you need anything.

On Tue, Jun 9, 2015, 10:49 PM Darius Scott notifications@github.com wrote:

> Figured it was something simple.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/Civcraft/CivChat2/pull/12#issuecomment-110577490.

suirad committed:
Finalized PM's; Cleaned up imports and warnings
Pms are pretty much finished. I managed to get the chat channels working cross server, so now you can do:
/msg suiradx0 Hello
---
/msg suiradx0
Hello

suirad commented:
Another commit. So i was able to finish pms and make them Mercury.enabled aware. I also made chatchannels work properly with cross server pms. I would consider PMs done and civtestable.(Yay!)
Now, as far as group chat is concerned, im afraid i would rather have someone more comfortable with how groups are handled in namelayer to go ahead and add that. I have a feeling me trying to figure it out would end up slowing down other plugins getting worked on. Mostly because as i tried testing the regular group commands, i ended up erroring out in the group manager in namelayer(now that i think about it, maybe i need citadel to make it work). So if there was any part i could task someone, it would be to finish sharding the group chat.

On the [Namelayer](https://github.com/Civcraft/NameLayer/pull/114) side, i made a slight change to make it track the server that each player is logged into. This should optimize things further when dealing with individual players. For example, in civchat i changed it to send a message directly to the server the player is logged into for a pm to another server. This is much better than the previous send to all servers when a pm was sent through.

Last thing done was clean out unused imports and static warnings.

@ttk2 , mind giving me the go ahead to start working on another plugin? If so, please make an issue on the respective repo, detail the features wanted added sharding wise, and either assign it to me if you can, or just message me the link somewhere.

ttk2 commented:
Roruke is out for a few days so its probably better if you start on somthing not namelayer related, that means PrisonPearl, JukeAlert should be super simple but it does require knowing what groups are where. 

I will go and spec out PP changes in a few minutes. Otherwise this can wait until someone can review it and then we can merge and start testing. 

BTW we should probably get venus under the Civcraft github org.

jjj5311 commented:
@ttk2 this looks good to me

jjj5311 commented on src/vg/civcraft/mc/civchat2/command/commands/GroupChat.java:
```diff
@@ -61,7 +61,7 @@ public boolean execute(CommandSender sender, String[] args){
 			}
 		}
-		Group group = gm.getGroup(args[0]);
+		Group group = GroupManager.getGroup(args[0]);
```
@suirad  is there a benefit to not using gm.getGroup? if so probably should just delete the initialization https://github.com/Civcraft/CivChat2/pull/12/files#diff-a416eb617fde805bb3d1e79c0d75ffffR45

ttk2 commented:
oh drat, we need the CLA on this repo still. No other comments on the code jjj?

jjj5311 commented:
Nope looks good to me

On Wednesday, June 10, 2015, ttk2 notifications@github.com wrote:

> oh drat, we need the CLA on this repo still. No other comments on the code
> jjj?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/Civcraft/CivChat2/pull/12#issuecomment-110955662.

ttk2 commented:
everyone agrees to a permissive license?  (Sorry no CLA setup yet)

suirad commented:
@Jjj5311 the change was part of the cleanup of things not being used
statically. Yea the gm init could probably be removed.

everyone agrees to a permissive license? (Sorry no CLA setup yet)

—
Reply to this email directly or view it on GitHub
https://github.com/Civcraft/CivChat2/pull/12#issuecomment-110962099.

ttk2 commented:
merged, so we should be ready to go and test things out, should probably have an issue for Citadel group chat still not working though.

suirad commented:
Right. Hopefully someone goes ahead an picks that up. NameLayer should
probably have a depend on citadel or at least a softdepend and a check
before doing group stuff. I wasn't around when group chat was added(haven't
played in over a year, but have been keeping up with changes and the sub on
and off), so I wasn't 100% of the mechanics.

On Thu, Jun 11, 2015 at 3:00 PM, ttk2 notifications@github.com wrote:

> merged, so we should be ready to go and test things out, should probably
> have an issue for Citadel group chat still not working though.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/Civcraft/CivChat2/pull/12#issuecomment-111106222.

ttk2 commented:
All groups are in namelayer. Citadel just handles the reinforcement db and
breaks. It defers to namelayer for groups.

On Thu, Jun 11, 2015, 7:19 AM Darius Scott notifications@github.com wrote:

> Right. Hopefully someone goes ahead an picks that up. NameLayer should
> probably have a depend on citadel or at least a softdepend and a check
> before doing group stuff. I wasn't around when group chat was added(haven't
> played in over a year, but have been keeping up with changes and the sub on
> and off), so I wasn't 100% of the mechanics.
> 
> On Thu, Jun 11, 2015 at 3:00 PM, ttk2 notifications@github.com wrote:
> 
> > merged, so we should be ready to go and test things out, should probably
> > have an issue for Citadel group chat still not working though.
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > https://github.com/Civcraft/CivChat2/pull/12#issuecomment-111106222.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/Civcraft/CivChat2/pull/12#issuecomment-111112728.

suirad commented:
Interesting. I should probably read the code more.

On Thu, Jun 11, 2015 at 3:52 PM, ttk2 notifications@github.com wrote:

> All groups are in namelayer. Citadel just handles the reinforcement db and
> breaks. It defers to namelayer for groups.
> 
> On Thu, Jun 11, 2015, 7:19 AM Darius Scott notifications@github.com
> wrote:
> 
> > Right. Hopefully someone goes ahead an picks that up. NameLayer should
> > probably have a depend on citadel or at least a softdepend and a check
> > before doing group stuff. I wasn't around when group chat was
> > added(haven't
> > played in over a year, but have been keeping up with changes and the sub
> > on
> > and off), so I wasn't 100% of the mechanics.
> > 
> > On Thu, Jun 11, 2015 at 3:00 PM, ttk2 notifications@github.com wrote:
> > 
> > > merged, so we should be ready to go and test things out, should
> > > probably
> > > have an issue for Citadel group chat still not working though.
> > > 
> > > —
> > > Reply to this email directly or view it on GitHub
> > > https://github.com/Civcraft/CivChat2/pull/12#issuecomment-111106222.
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > https://github.com/Civcraft/CivChat2/pull/12#issuecomment-111112728.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/Civcraft/CivChat2/pull/12#issuecomment-111122409.

ttk2 commented:
I know most things, so ask if you need advice. You can also request my presence on mumble if you need it.
